pkg/controllers/provisioning/scheduling/volumetopology.go

```diff
@@ -76,30 +76,18 @@ func (v *VolumeTopology) Inject(ctx context.Context, pod *v1.Pod) error {
 }
 
 func (v *VolumeTopology) getRequirements(ctx context.Context, pod *v1.Pod, volume v1.Volume) ([]v1.NodeSelectorRequirement, error) {
-	defaultStorageClassName, err := scheduling.DiscoverDefaultStorageClassName(ctx, v.kubeClient)
+	pvc, err := volumeutil.GetPersistentVolumeClaim(ctx, v.kubeClient, pod, volume)
 	if err != nil {
-		return nil, fmt.Errorf("discovering default storage class, %w", err)
+		return nil, fmt.Errorf("discovering persistent volume claim, %w", err)
 	}
-
-	// Get VolumeName and StorageClass name from PVC
-	pvc := &v1.PersistentVolumeClaim{}
-	switch {
-	case volume.PersistentVolumeClaim != nil:
-		if err = v.kubeClient.Get(ctx, types.NamespacedName{Namespace: pod.Namespace, Name: volume.PersistentVolumeClaim.ClaimName}, pvc); err != nil {
-			return nil, fmt.Errorf("discovering persistent volume claim, %w", err)
-		}
-	case volume.Ephemeral != nil:
-		// generated name per https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#persistentvolumeclaim-naming
-		if err = v.kubeClient.Get(ctx, types.NamespacedName{Namespace: pod.Namespace, Name: fmt.Sprintf("%s-%s", pod.Name, volume.Name)}, pvc); err != nil {
-			return nil, fmt.Errorf("discovering persistent volume claim for ephemeral volume, %w", err)
-		}
-	default:
+	// may not have a pvc
+	if pvc == nil {
 		return nil, nil
 	}
+
+	// Get VolumeName and StorageClass name from PVC
 	storageClassName := lo.FromPtr(pvc.Spec.StorageClassName)
```

Comment:
Probably not, no. I can go ahead and tear that out 👍

Reply:
Actually I'm going to backpedal here, the storage class could still be `nil` if there was no default storage class when the PVC was created. If a default exists when the PVC is created it will be added via an admission controller, otherwise it can be reconciled on later.